.github/workflows/diff_report.yml

```diff
@@ -0,0 +1,150 @@
+      config_link: ${{ steps.parse.outputs.config_link }}
+      user: ${{ steps.branch.outputs.user }}
+      branch: ${{ steps.branch.outputs.ref }}
+      
+    steps:
+    
+     - name: Issue comment action
+       if: github.event_name == 'issue_comment'
+       run: |
+        echo "${{github.event.issue.body}}" > text
+        echo "${{github.event.issue.user.login}}" > user
+        wget -q ${{github.event.issue.pull_request.url}} -O info.json
+        jq .head.ref info.json > branch
+       
+     - name: PR action
+       if: github.event_name == 'pull_request'
+       run: |
+        echo "${{github.event.pull_request.body}}" > text
+        echo ${{github.event.pull_request.user.login}} > user
+        echo ${{github.event.pull_request.head.ref}} > branch
+        
+     - name: Parse body
+       id: parse
+       run: |
+        grep "^Diff Regression projects:" text > temp || echo "" > temp
+        sed 's/Diff Regression projects: //' temp > proj
+        echo ::set-output name=proj_link::$(cat proj)
+        grep "^Diff Regression config:" text > temp || echo "" > temp
+        sed 's/Diff Regression config: //' temp > config
+        echo ::set-output name=config_link::$(cat config)
+        
+     - name: Set branch and head_label
+       id: branch
+       run: |
+        echo ::set-output name=user::$(cat user)
+        echo ::set-output name=ref::$(cat branch)
+      
+     - name: info
+       run: |
+        cat user
+        cat branch
+        echo ${{steps.branch.outputs.ref}}
+        echo ${{steps.branch.outputs.user}}:
+     
+  make_report:
+    runs-on: ubuntu-latest
+    needs: parse_body
+    if: needs.parse_body.outputs.proj_link != '' && needs.parse_body.outputs.config_link != ''
+    steps: 
+       
+     - name: Download files
+       run: |
+        wget -q ${{needs.parse_body.outputs.proj_link}} -O proj.properties
+        wget -q ${{needs.parse_body.outputs.config_link}} -O config.xml
+        
+     - name: Download checkstyle
+       uses: actions/checkout@v2
+       with:
+        repository: ${{needs.parse_body.outputs.user}}/checkstyle
+        ref: master
+        path: checkstyle
+        fetch-depth: 0
+        
+     - name: Check branches
+       run: |
+         cd checkstyle
+         git branch -r
+         cd ../
+     
+     - name: Download contribution
+       uses: actions/checkout@v2
+       with:
+        repository: checkstyle/contribution
+        path: contribution
+         
+     - name: Print branch
+       run: |
+         echo ${{needs.parse_body.outputs.branch}}
+         
+     - name: Relocate .xml and .properties 
+       run: |
+         mv config.xml ./contribution/checkstyle-tester/
+         mv proj.properties ./contribution/checkstyle-tester/
+         
+     - name: Generate report
+       run: |
+         bash
+         REF="origin/${{needs.parse_body.outputs.branch}}"
+         cd contribution/checkstyle-tester
+         sudo apt-get install mercurial
```

Comment:
please remove this, mercurial is used only for old jdk repos, right now all of them are on github.
so such install is not required.